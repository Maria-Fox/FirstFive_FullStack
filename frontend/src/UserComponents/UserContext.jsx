import React from "react";

// Create user context instance to later provide in App.js and use throughout application. Holds auth user

const UserContext = React.createContext();

export default UserContext;