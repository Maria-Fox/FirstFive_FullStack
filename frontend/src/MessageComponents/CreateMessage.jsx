import React, { useState, useContext } from "react";
import { useParams, useNavigate } from "react-router-dom";
import API from "../API";
import AlertNotification from "../Common/AlertNotifications";
import UserContext from "../UserComponents/UserContext";
import { Form, Label, Input, Button, Card } from "reactstrap"

const CreateMessage = () => {

  // message from, message_to
  const { to_username } = useParams();

  // ***************************************************************

  let initial_state = {
    message_to: to_username,
    body: ""
  };

  // ***************************************************************

  const [msgData, setMsgData] = useState(initial_state);
  const [errors, setErrors] = useState(null);
  const navigate = useNavigate();
  const { authUser } = useContext(UserContext);

  const handleChange = async (e) => {
    let { name, value } = e.target;

    setMsgData(msgData => ({
      ...msgData,
      [name]: value
    }));
  };

  // ***************************************************************


  const handleSubmit = async (e) => {
    try {
      e.preventDefault();
      await API.createMessage(authUser, { message_from: authUser, ...msgData });
      navigate(`/messages/${authUser}/all`);
    } catch (e) {
      setErrors(e);
      return;
    };
  };


  // ***************************************************************


  return (
    <>
      <h1 style={{ textAlign: "center" }}>New Message</h1>
      <Card>

        {errors ? <AlertNotification messages={errors} /> : null}

        <Form onSubmit={handleSubmit} >

          <p style={{ color: "navy", fontSize: "1.5rem" }}>Message to: {to_username}</p>

          <Label htmlFor="body" >
            <Input
              type="textarea"
              id="body"
              value={msgData.body}
              name="body"
              placeholder="Hi - I am interested in hearing more about your project. I have frontend expereince. Can you use a new project member?"
              required
              cols="100"
              rows="10"
              onChange={handleChange}
            >
            </Input>
          </Label>

          <br />
          <Button>Send</Button>

        </Form>
      </Card >
    </>
  )
};

export default CreateMessage;