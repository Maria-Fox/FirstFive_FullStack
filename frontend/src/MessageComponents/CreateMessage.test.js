import React from "react";
import { MemoryRouter } from "react-router-dom";
import CreateMessage from "./CreateMessage";
import { render, getByText } from "@testing-library/react";
import UserContext from "../UserComponents/UserContext";


test("Smoke - renders the CreateMessageForm", function () {

  const authUser = { username: "softwareDev1" };

  render(
    <MemoryRouter>
      <UserContext.Provider value={authUser}>
        <CreateMessage />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

test("Snapshot- CreateMsg forms renders for valid user", function () {

  const authUser = { username: "softwareDev1" }
  let { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={authUser}>
        <CreateMessage />
      </UserContext.Provider>
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});

