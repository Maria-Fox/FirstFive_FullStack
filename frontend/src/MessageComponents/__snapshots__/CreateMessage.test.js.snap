// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot- CreateMsg forms renders for valid user 1`] = `
<DocumentFragment>
  <h1
    style="text-align: center;"
  >
    New Message
  </h1>
  <div
    class="card"
  >
    <form
      class=""
    >
      <p
        style="color: navy; font-size: 1.5rem;"
      >
        Message to: 
      </p>
      <label
        class="form-label"
        for="body"
      >
        <textarea
          class="form-control"
          cols="100"
          id="body"
          name="body"
          placeholder="Hi - I am interested in hearing more about your project. I have frontend expereince. Can you use a new project member?"
          required=""
          rows="10"
        />
      </label>
      <br />
      <button
        class="btn btn-secondary"
      >
        Send
      </button>
    </form>
  </div>
</DocumentFragment>
`;
